scripts/Layer.gd

```diff
@@ -30,3 +44,15 @@ func update_texture() -> void:
 	if changed:
 		texture.set_data(image)
 		changed = false
+		update_preview()
+
+# Updates the preview image for each layer from the canvas
+func update_preview() -> void:
+	preview_image.fill(Color.white)
+	preview_image.blend_rect(image, Rect2(Vector2.ZERO, size), Vector2.ZERO)
```

Comment:
For larger images this might get expensive, not sure though. Maybe just set the texture data to the layer image as a copy and then  in the ui just have a white square underneath that doesn't change. This would remove the need to constantly blend rect.